sycl/doc/design/spirv-extensions/SPV_INTEL_subgroup_requirements.asciidoc

```diff
@@ -0,0 +1,261 @@
+= SPV_INTEL_subgroup_requirements
+
+== Name Strings
+
+SPV_INTEL_subgroup_requirements
+
+== Contact
+
+To report problems with this extension, please open a new issue at:
+
+https://github.com/KhronosGroup/SPIRV-Registry
+
+== Contributors
+
+// spell-checker: disable
+* Ben Ashbaugh, Intel
+* Pekka Jääskeläinen, Intel
+* Henry Linjamäki, Intel
+* John Pennycook, Intel
+// spell-checker: enable
+
+== Notice
+
+Copyright (c) 2023 Intel Corporation.  All rights reserved.
+
+== Status
+
+* Working Draft
+
+This is a preview extension specification, intended to provide early access to a
+feature for review and community feedback.
+When the feature matures, this specification may be released as a formal
+extension.
+
+Because the interfaces defined by this specification are not final and are
+subject to change they are not intended to be used by shipping software
+products.
+If you are interested in using this feature in your software product, please let
+us know!
+
+== Version
+
+[width="40%",cols="25,25"]
+|========================================
+| Last Modified Date | {docdate}
+| Revision           | C
+|========================================
+
+== Dependencies
+
+This extension is written against the SPIR-V Specification, Version 1.6 Revision 2.
+
+This extension requires SPIR-V 1.0.
+
+== Overview
+
+This extension allows a compiler generating SPIR-V to add additional
+subgroup requirements to entry points (kernels) in a SPIR-V module.
+These requirements enable programmers and compilers to reason better about how
+subgroups behave for an entry point.
+
+Specifically, this extension adds two new execution modes:
+
+* The ability to specify a symbolic *NamedSubgroupSizeINTEL* subgroup size for
+an entry point.
+* The ability to specify the *SubgroupLaneMappingINTEL* for an entry point.
+The subgroup lane mapping defines how the invocations (work-items) in a
+workgroup are assigned to subgroups.
+
+== Extension Name
+
+To use this extension within a SPIR-V module, the following *OpExtension* must
+be present in the module:
+
+----
+OpExtension "SPV_INTEL_subgroup_requirements"
+----
+
+== Modifications to the SPIR-V Specification, Version 1.6
+
+=== Validation Rules
+
+Add validation rules to section 2.16.1 Universal Validation Rules under Entry Point:
+
+* Each *OpEntryPoint* must contain at most one of the *SubgroupSize* or
+*NamedSubgroupSizeINTEL* execution modes.
+* Each *OpEntryPoint* must contain at most one of the *SubgroupLaneMappingINTEL*
+execution modes.
+
+=== Execution Mode
+
+Modify Section 3.6, "Execution Mode", adding these rows to the Execution Mode table:
+
+--
+[cols="^4,20,3*5,22",options="header"]
+|=====
+2+^.^| Execution Mode 3+<.^| Extra Operands | Enabling Capabilities
+| 6446 | *NamedSubgroupSizeINTEL*
+   3+| _Named Subgroup Size_ | *SubgroupRequirementsINTEL*
```

Comment:
The `SubgroupSize` execution mode has `SubgroupDispatch` as an enabling capability, should `NamedSubgroupSizeINTEL` have `SubgroupDispatch` as well?

Reply:
Good question, and good observation!

I think it could, but currently the `SubgroupDispatch` capability "implicitly declares" the `DeviceEnqueue` capability.  This is almost certainly a mistake, and we're looking at ways to fix this in the Khronos SPIR working group (for Khronos folks, see internal SPIR-V issue 525).  Until this is resolved though, I'd prefer not to propagate this mistake to another feature, and I think this feature can stand on its own without needing anything else from `SubgroupDispatch`.  Does this make sense?

Reply:
I see, that makes sense, thanks for the clarification.